File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/model/Card.java
package com.lattis.ellipse.domain.model;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by ssd3 on 7/26/17.
 */

public class Card implements Parcelable {
    private int id;
    private int user_id;
    private String stripe_net_profile_id;
    private String stripe_net_payment_id;
    private boolean is_primary;
    private String type_card;
    private String cc_no;
    private int exp_month;
    private int exp_year;
    private String fingerprint;
    private String cc_type;
    private int created_date;
    private String last_updated;


    protected Card(Parcel in) {
        id = in.readInt();
        user_id = in.readInt();
        stripe_net_profile_id = in.readString();
        stripe_net_payment_id = in.readString();
        is_primary = in.readByte() != 0;
        type_card = in.readString();
        cc_no = in.readString();
        exp_month = in.readInt();
        exp_year = in.readInt();
        fingerprint = in.readString();
        cc_type = in.readString();
        created_date = in.readInt();
        last_updated = in.readString();
    }

    public Card() {

    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeInt(user_id);
        dest.writeString(stripe_net_profile_id);
        dest.writeString(stripe_net_payment_id);
        dest.writeByte((byte) (is_primary ? 1 : 0));
        dest.writeString(type_card);
        dest.writeString(cc_no);
        dest.writeInt(exp_month);
        dest.writeInt(exp_year);
        dest.writeString(fingerprint);
        dest.writeString(cc_type);
        dest.writeInt(created_date);
        dest.writeString(last_updated);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Card> CREATOR = new Creator<Card>() {
        @Override
        public Card createFromParcel(Parcel in) {
            return new Card(in);
        }

        @Override
        public Card[] newArray(int size) {
            return new Card[size];
        }
    };

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public String getStripe_net_profile_id() {
        return stripe_net_profile_id;
    }

    public void setStripe_net_profile_id(String stripe_net_profile_id) {
        this.stripe_net_profile_id = stripe_net_profile_id;
    }

    public String getStripe_net_payment_id() {
        return stripe_net_payment_id;
    }

    public void setStripe_net_payment_id(String stripe_net_payment_id) {
        this.stripe_net_payment_id = stripe_net_payment_id;
    }

    public boolean getIs_primary() {
        return is_primary;
    }

    public void setIs_primary(boolean is_primary) {
        this.is_primary = is_primary;
    }

    public String getType_card() {
        return type_card;
    }

    public void setType_card(String type_card) {
        this.type_card = type_card;
    }

    public String getCc_no() {
        return cc_no;
    }

    public void setCc_no(String cc_no) {
        this.cc_no = cc_no;
    }

    public int getExp_month() {
        return exp_month;
    }

    public void setExp_month(int exp_month) {
        this.exp_month = exp_month;
    }

    public int getExp_year() {
        return exp_year;
    }

    public void setExp_year(int exp_year) {
        this.exp_year = exp_year;
    }

    public String getFingerprint() {
        return fingerprint;
    }

    public void setFingerprint(String fingerprint) {
        this.fingerprint = fingerprint;
    }

    public String getCc_type() {
        return cc_type;
    }

    public void setCc_type(String cc_type) {
        this.cc_type = cc_type;
    }

    public int getCreated_date() {
        return created_date;
    }

    public void setCreated_date(int created_date) {
        this.created_date = created_date;
    }

    public String getLast_updated() {
        return last_updated;
    }

    public void setLast_updated(String last_updated) {
        this.last_updated = last_updated;
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/models/Card.kt
package com.lattis.domain.models

import android.os.Parcel
import android.os.Parcelable


class Card : Parcelable {
    var id: Int = 0
    var user_id: Int = 0
    var stripe_net_profile_id: String? = null
    var stripe_net_payment_id: String? = null
    var is_primary: Boolean = false
    var type_card: String? = null
    var cc_no: String? = null
    var exp_month: Int = 0
    var exp_year: Int = 0
    var fingerprint: String? = null
    var cc_type: String? = null
    var created_date: Int = 0
    var last_updated: String? = null
    var card_id : String?=null

    fun getIs_primary():Boolean{
        return is_primary;
    }


    protected constructor(`in`: Parcel) {
        id = `in`.readInt()
        user_id = `in`.readInt()
        stripe_net_profile_id = `in`.readString()
        stripe_net_payment_id = `in`.readString()
        is_primary = `in`.readByte().toInt() != 0
        type_card = `in`.readString()
        cc_no = `in`.readString()
        exp_month = `in`.readInt()
        exp_year = `in`.readInt()
        fingerprint = `in`.readString()
        cc_type = `in`.readString()
        created_date = `in`.readInt()
        last_updated = `in`.readString()
        card_id = `in`.readString()
    }

    constructor() {

    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeInt(user_id)
        dest.writeString(stripe_net_profile_id)
        dest.writeString(stripe_net_payment_id)
        dest.writeByte((if (is_primary) 1 else 0).toByte())
        dest.writeString(type_card)
        dest.writeString(cc_no)
        dest.writeInt(exp_month)
        dest.writeInt(exp_year)
        dest.writeString(fingerprint)
        dest.writeString(cc_type)
        dest.writeInt(created_date)
        dest.writeString(last_updated)
        dest.writeString(card_id)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Card> {
        override fun createFromParcel(parcel: Parcel): Card {
            return Card(parcel)
        }

        override fun newArray(size: Int): Array<Card?> {
            return arrayOfNulls(size)
        }
    }


}
